src/plugins/kibana_react/public/code_editor/code_editor.tsx

```diff
@@ -150,14 +145,15 @@ export class CodeEditor extends React.Component<Props, {}> {
     monaco.editor.defineTheme('euiColors', this.props.useDarkTheme ? DARK_THEME : LIGHT_THEME);
   };
 
-  _editorDidMount = (
-    editor: monacoEditor.editor.IStandaloneCodeEditor,
-    monaco: typeof monacoEditor
-  ) => {
+  _editorDidMount = (editor: monaco.editor.IStandaloneCodeEditor, __monaco: unknown) => {
+    if (__monaco !== monaco) {
+      throw new Error('react-monaco-editor is using a different version of monaco');
+    }
```

Comment:
It seems the only reason to expose the `monaco` instance to handlers like `editorDidMount()` is so consumers can get access to the version of monaco being used by `react-monaco-editor`. In our case this version should always be the same as the version imported from `@kbn/ui-shared-deps/monaco`, so I updated this and the signature for these handlers to ensure we keep a single source of truth.

Reply:
Excellent, yeah this makes sense to me